### PR TITLE
Add documentation about configurable input size

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
Add configurable input size documentation and update HPO documentation.
Fix minor configurable input size bug

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+This flexibility enables users to tailor the input size to their specific needs without manually altering
+the data pipeline configurations.
+
+To utilize this feature, simply specify the desired input size as an argument during the train command.
+Additionally, OTX ensures compatibility with model trained on non-default input sizes by automatically adjusting
+the data pipeline to match the input size during other engine entry points.
+
+Usage example:
+
+.. code-block::
```
Comment (harimkang) [inline docs/source/guide/explanation/additional_features/configurable_input_size.rst:15]:
```suggestion
.. code-block:: shell
```

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+The Adaptive Input Size feature intelligently determines an optimal input size for the model
+by analyzing the dataset's statistics.
+It operates in two distinct modes: "auto" and "downscale".
+In "auto" mode, the input size may increase or decrease based on the dataset's characteristics.
+In "downscale" mode, the input size will either decrease or remain unchanged, ensuring that the model training or inference speed deosn't drop.
+
+
+To activate this feature, use the following command with the desired mode:
+
+.. code-block::
```
Comment (harimkang) [inline docs/source/guide/explanation/additional_features/configurable_input_size.rst]:
```suggestion
.. code-block:: shell
```

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+Additionally, OTX ensures compatibility with model trained on non-default input sizes by automatically adjusting
+the data pipeline to match the input size during other engine entry points.
+
+Usage example:
+
+.. code-block::
+
+    $ otx train \
+        --config ... \
+        --data.input_size 512
```
Comment (harimkang) [inline docs/source/guide/explanation/additional_features/configurable_input_size.rst]: Couldn't we provide the same example via the API? How about adding API examples as well?

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+In "downscale" mode, the input size will either decrease or remain unchanged, ensuring that the model training or inference speed deosn't drop.
+
+
+To activate this feature, use the following command with the desired mode:
+
+.. code-block::
+
+    $ otx train \
+        --config ... \
+        --data.adaptive_input_size "auto | downscale"
```
Comment (harimkang) [inline docs/source/guide/explanation/additional_features/configurable_input_size.rst]: How about adding API examples?

**Review by goodsong81 — DISMISSED** [review]

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst (hunk 1 above)
Comment (eunwoosh) [reply docs/source/guide/explanation/additional_features/configurable_input_size.rst:15]: done https://github.com/openvinotoolkit/training_extensions/pull/3870/commits/400bd294597d4102c5fd414376c0d4859918c97f

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst (hunk 2 above)
Comment (eunwoosh) [reply docs/source/guide/explanation/additional_features/configurable_input_size.rst]: done https://github.com/openvinotoolkit/training_extensions/pull/3870/commits/400bd294597d4102c5fd414376c0d4859918c97f

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst (hunk 3 above)
Comment (eunwoosh) [reply docs/source/guide/explanation/additional_features/configurable_input_size.rst]: done https://github.com/openvinotoolkit/training_extensions/pull/3870/commits/400bd294597d4102c5fd414376c0d4859918c97f

File: docs/source/guide/explanation/additional_features/configurable_input_size.rst (hunk 4 above)
Comment (eunwoosh) [reply docs/source/guide/explanation/additional_features/configurable_input_size.rst]: done https://github.com/openvinotoolkit/training_extensions/pull/3870/commits/400bd294597d4102c5fd414376c0d4859918c97f

**Review by harimkang — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]

**Pull request merged**